### PR TITLE
[Messenger] Document middleware factories

### PR DESCRIPTION
**Pull request by ogizanagi**
Refs:

- Symfony PR: https://github.com/symfony/symfony/pull/27128
- DoctrineBundle PR: https://github.com/doctrine/DoctrineBundle/pull/817

### CODE REVIEW

File: messenger.rst [hunk 1]
```diff
@@ -273,6 +273,114 @@ you can disable them like this:
                 messenger.bus.default:
                     default_middleware: false
 
+Using middleware factories
```
Comment (xabbuh) [inline messenger.rst]: Using Middleware Factories

Comment (ogizanagi) [reply messenger.rst]: Fixed thanks

File: messenger.rst [hunk 2]
```diff
@@ -273,6 +273,116 @@ you can disable them like this:
                 messenger.bus.default:
                     default_middleware: false
 
+Using Middleware Factories
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Sometimes middleware are configurable using factories. There are two types of
```
Comment (ogizanagi) [inline messenger.rst]: Turned like this, it seems to me the sentence seems is saying say there are two different ways to achieve configurable middleware, while it's actually two required services definitions. One registering the factory class as a service. The second one registering an abstract definition building the service through the first factory.

**Review by ogizanagi — COMMENTED** [review]
Thanks a lot @javiereguiluz for improving this PR ❤️

File: messenger.rst (hunk 2 above)
Comment (javiereguiluz) [reply messenger.rst]: I'm sorry for having changed the meaning of the docs. I've reworded it again to follow your explanation. Cheers!

**Comment by javiereguiluz** [comment]
Thanks Maxime.

**Pull request merged**